### PR TITLE
Day view page fetch task from backend

### PR DESCRIPTION
**Pull request by PeterOu8**
Implemented the fetch task due today feature. Implemented the fetch function to get all tasks due by today's date.

### CODE REVIEW

File: blotztask-ui/src/app/task-dayview/page.tsx [hunk 1]
```diff
@@ -1,62 +1,44 @@
 'use client';
 
-import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card';
-import { Checkbox } from '@/components/ui/checkbox';
+import { Card, CardContent, CardHeader } from '@/components/ui/card';
+// import { Checkbox } from '@/components/ui/checkbox';
```
Comment (Ben0189) [inline blotztask-ui/src/app/task-dayview/page.tsx:4]: remove?

File: blotztask-ui/src/services/taskService.ts [hunk 2]
```diff
@@ -18,4 +19,26 @@ export const fetchAllTaskItems = async (): Promise<TaskItemDTO[]> => {
...
+      {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }
+    );
+  
+    if (!response.ok) {
+      throw new Error('Network response was not ok');
```
Comment (Ben0189) [inline blotztask-ui/src/services/taskService.ts:39]:
```suggestion
      throw new Error('Error in fetching Task Item from backend');
```

File: blotztask-ui/tailwind.config.ts [hunk 3]
```diff
@@ -21,6 +21,11 @@ const config: Config = {
         'primary-dark': '#2C3233',
         'secondary': '#278291',
         'warn': '#F42F67',
+        'checkbox': '#989ca4',
```
Comment (Ben0189) [inline blotztask-ui/tailwind.config.ts:24]: what is the checkbox for?

**Review by Ben0189 — CHANGES_REQUESTED** [review]

File: blotztask-ui/src/app/task-dayview/page.tsx (hunk 1 above)
Comment (PeterOu8) [reply blotztask-ui/src/app/task-dayview/page.tsx:4]: The checkbox will be applied per figma prototype. I tried implement this but realised it was in another ticket.

**Review by Ben0189 — DISMISSED** [review]

File: blotztask-ui/tailwind.config.ts (hunk 3 above)
Comment (PeterOu8) [reply blotztask-ui/tailwind.config.ts:24]: I have the checkbox component wrote out already but I did not take the ticket. I tried styling the shadcn checkbox component therefore there is another color.

**Review by Ben0189 — APPROVED** [review]

**Pull request merged**